go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -1184,3 +1185,146 @@ func (bnc *BaseNetworkController) GetSamplingConfig() *libovsdbops.SamplingConfi
+	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLAdvertisedUDN, advertisedUDNIsolationACLID,
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey: advertisedUDNSubnetsCtrl,
+		})
+}
+
+func GetAdvertisedUDNSubnetsAcceptACLdbIDs(networkName string) *libovsdbops.DbObjectIDs {
+	return libovsdbops.NewDbObjectIDs(libovsdbops.ACLAdvertisedUDN, advertisedUDNIsolationACLID,
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey: networkName,
+		})
+}
+
+func BuildAdvertisedUDNSubnetsDropACL(advertisedUDNSubnetsAddressSet addressset.AddressSet) *nbdb.ACL {
+	var dropMatches []string
+	v4AddrSet, v6AddrSet := advertisedUDNSubnetsAddressSet.GetASHashNames()
+	if v4AddrSet != "" {
+		dropMatches = append(dropMatches, fmt.Sprintf("(ip4.src == $%s && ip4.dst == $%s)", v4AddrSet, v4AddrSet))
+	}
+	if v6AddrSet != "" {
+		dropMatches = append(dropMatches, fmt.Sprintf("(ip6.src == $%s && ip6.dst == $%s)", v6AddrSet, v6AddrSet))
+	}
+
+	dropACL := libovsdbutil.BuildACL(
+		GetAdvertisedUDNSubnetsDropACLdbIDs(),
+		types.DefaultDenyPriority,
+		strings.Join(dropMatches, " || "),
+		nbdb.ACLActionDrop,
+		nil,
+		libovsdbutil.LportEgress)
+	dropACL.Tier = types.PrimaryACLTier
+	return dropACL
+}
+
+func (bnc *BaseNetworkController) addAdvertisedUDNIsolation(nodeName string) error {
+	var acceptMatches, cidrs []string
+	var ops []ovsdb.Operation
+
+	addrSet, err := bnc.addressSetFactory.GetAddressSet(GetAdvertisedUDNSubnetsAddressSetDBIDs())
+	if err != nil {
+		return err
+	}
+
+	for _, subnet := range bnc.Subnets() {
+		ipPrefix := "ip4"
+		if utilnet.IsIPv6CIDR(subnet.CIDR) {
+			ipPrefix = "ip6"
+		}
+		acceptMatches = append(acceptMatches, fmt.Sprintf("(%s.src == %s && %s.dst == %s)", ipPrefix, subnet.CIDR, ipPrefix, subnet.CIDR))
+		cidrs = append(cidrs, subnet.CIDR.String())
+
+	}
+
+	addrOps, err := addrSet.AddAddressesReturnOps(cidrs)
+	if err != nil {
+		return fmt.Errorf("failed to add addresses %q to the address set: %w", cidrs, err)
+	}
+	ops = append(ops, addrOps...)
+
+	if len(acceptMatches) > 0 {
+		acceptACL := libovsdbutil.BuildACL(
+			GetAdvertisedUDNSubnetsAcceptACLdbIDs(bnc.GetNetworkName()),
+			types.DefaultAllowPriority,
+			strings.Join(acceptMatches, " || "),
+			nbdb.ACLActionPass,
+			nil,
+			libovsdbutil.LportEgress)
+		acceptACL.Tier = types.PrimaryACLTier
+
+		ops, err = libovsdbops.CreateOrUpdateACLsOps(bnc.nbClient, ops, nil, acceptACL)
+		if err != nil {
+			return fmt.Errorf("failed to create or update udn isolation accept ACL %w", err)
+		}
+		ops, err = libovsdbops.AddACLsToLogicalSwitchOps(bnc.nbClient, ops, bnc.GetNetworkScopedSwitchName(nodeName), acceptACL)
+		if err != nil {
+			return fmt.Errorf("failed to add udn isolation accept ACL to switch %s: %w", bnc.GetNetworkScopedSwitchName(nodeName), err)
+		}
+	}
+
+	dropACL := BuildAdvertisedUDNSubnetsDropACL(addrSet)
+	ops, err = libovsdbops.CreateOrUpdateACLsOps(bnc.nbClient, ops, nil, dropACL)
```

Comment:
ah I see so N ACLs 1 per advertised UDN, 1ACL global deny,
so N+1 ACLs across N switches is that correct?
so its per network N+1 ACLs? N(N+1) ACLs?

Reply:
there is one global drop ACL and one per-network ACL.

Reply:
soooo its 2ACLs for each network actually right?
so if I have N networks its 2N ACLs on each node correct?